tests/test_utils_mean.py

```diff
@@ -1,134 +1,127 @@
+    ],
+)
+def test_mean_shape(mean):
+    """Test the shape of mean"""
+    covmats, _, A = generate_cov(5, 3)
+    C = mean(covmats)
+    assert C.shape == (3, 3)
 
 
-def test_riemann_mean_with_init():
-    """Test the riemannian mean with init"""
-    covmats, diags, A = generate_cov(100, 3)
-    C = mean_riemann(covmats, init=covmats[0])
-    Ctrue = np.exp(np.log(diags).mean(0))
-    Ctrue = np.dot(np.dot(A, np.diag(Ctrue)), A.T)
-    assert_array_almost_equal(C, Ctrue)
+@pytest.mark.parametrize("mean", [mean_riemann, mean_logdet])
+def test_mean_shape_with_init(mean):
+    """Test the shape of mean with init"""
+    n_trials, n_channels = 5, 3
+    covmats, _, A = generate_cov(n_trials, n_channels)
+    C = mean(covmats, init=covmats[0])
+    assert C.shape == (n_channels, n_channels)
 
 
-def test_logeuclid_mean():
-    """Test the logeuclidean mean"""
-    covmats, _, A = generate_cov(100, 3)
-    C = mean_logeuclid(covmats)
-    assert C.shape == (3, 3)
+@pytest.mark.parametrize("init", [True, False])
+def test_riemann_mean(init):
+    """Test the riemannian mean"""
+    n_trials, n_channels = 100, 3
+    covmats, diags, A = generate_cov(n_trials, n_channels)
+    if init:
+        C = mean_riemann(covmats, init=covmats[0])
+    else:
+        C = mean_riemann(covmats)
+    Ctrue = np.exp(np.log(diags).mean(0))
+    Ctrue = A @ np.diag(Ctrue) @ A.T
+    assert C == approx(Ctrue)
 
 
 def test_euclid_mean():
     """Test the euclidean mean"""
     covmats, _, _ = generate_cov(100, 3)
     C = mean_euclid(covmats)
-    assert C.shape == (3, 3)
-    assert_array_almost_equal(C, covmats.mean(axis=0))
+    assert C == approx(covmats.mean(axis=0))
 
 
 def test_identity_mean():
     """Test the logdet mean"""
     covmats, _, _ = generate_cov(100, 3)
     C = mean_identity(covmats)
-    assert C.shape == (3, 3)
-    assert_array_equal(C, np.eye(3))
-
-
-def test_logdet_mean():
-    """Test the logdet mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_logdet(covmats)
-    assert C.shape == (3, 3)
-
-
-def test_logdet_mean_with_init():
-    """Test the logdet mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_logdet(covmats, init=covmats[0])
-    assert C.shape == (3, 3)
+    assert np.all(C == np.eye(3))
 
 
-def test_ald_mean():
-    """Test the Ale mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_ale(covmats)
-    assert C.shape == (3, 3)
+def test_alm_mean():
+    """Test the ALM mean"""
+    covmats, _, _ = generate_cov(3, 3)
+    C_alm = mean_alm(covmats)
+    C_riem = mean_riemann(covmats)
+    assert C_alm == approx(C_riem)
 
 
-def test_kullback_mean():
-    """Test the kullback mean"""
-    covmats, _, _ = generate_cov(100, 3)
-    C = mean_kullback_sym(covmats)
+def test_alm_mean_maxiter():
+    """Test the ALM mean with max iteration"""
+    covmats, _, _ = generate_cov(3, 3)
+    C = mean_alm(covmats, maxiter=1, verbose=True)  # maxiter reached
     assert C.shape == (3, 3)
```

Comment:
Shape is already tested in `test_mean_shape`, isn't it?

Reply:
not with maxiter set